### ISSUE TITLE
Allow preview of index.xhtml as directory index page

### ISSUE THREAD

**Issue by jumapico** [nanoc/nanoc#986]
### Steps to reproduce

1. Generate an *index.xhtml* directory index file instead of *index.html*
2. Preview the site using *nanoc view* command
3. Go to the url 'http://127.0.0.1:3000'

### Expected behavior

The index.xhtml file is served

### Actual behavior

A *File not found: /* message is shown.

### Details

Many sites allows the usage of *index.xhtml* for the directory index page.

Particularly, in bitbucket pages and github pages the usage of index.xhtml for directory index is enabled and also in many web servers has that configuration enabled by default (checked for apache2 and lighttpd for example).

I would like if this can be enabled for the server that is started with the *nanoc view* command, avoiding the need of configure a web server for preview the generated site.

Maybe an option for add the index.xhtml page can be:

```diff
diff --git a/lib/nanoc/cli/commands/view.rb b/lib/nanoc/cli/commands/view.rb
index f625755..fdda17f 100644
--- a/lib/nanoc/cli/commands/view.rb
+++ b/lib/nanoc/cli/commands/view.rb
@@ -44,7 +44,9 @@ module Nanoc::CLI::Commands
         use Rack::ShowExceptions
         use Rack::Lint
         use Rack::Head
-        use Adsf::Rack::IndexFileFinder, root: site.config[:output_dir]
+        use Adsf::Rack::IndexFileFinder,
+          root: site.config[:output_dir],
+          index_filenames: %w( index.html index.xhtml )
         run Rack::File.new(site.config[:output_dir])
       end.to_app
```

Another option can be allow the **-i** and **--index-filenames** options in the *nanoc view* command for specify the index filenames (the same options used in the *adsf* command).

```diff
diff --git a/lib/nanoc/cli/commands/view.rb b/lib/nanoc/cli/commands/view.rb
index f625755..5c8882b 100644
--- a/lib/nanoc/cli/commands/view.rb
+++ b/lib/nanoc/cli/commands/view.rb
@@ -9,6 +9,8 @@ EOS
 required :H, :handler, 'specify the handler to use (webrick/mongrel/...)'
 required :o, :host,    'specify the host to listen on (default: 0.0.0.0)'
 required :p, :port,    'specify the port to listen on (default: 3000)'
+required :i, :'index-filenames',
+  'specify index filenames (comma-separated, default: index.html)'
 
 module Nanoc::CLI::Commands
   class View < ::Nanoc::CLI::CommandRunner
@@ -37,6 +39,13 @@ module Nanoc::CLI::Commands
         end
       end
 
+      # Get index filenames
+      if options.key?(:'index-filenames')
+        index_filenames = options[:'index_filenames'].split(',')
+      else
+        index_filenames = %w( index.html )
+      end
+
       # Build app
       site = self.site
       app = Rack::Builder.new do
@@ -44,7 +53,9 @@ module Nanoc::CLI::Commands
         use Rack::ShowExceptions
         use Rack::Lint
         use Rack::Head
-        use Adsf::Rack::IndexFileFinder, root: site.config[:output_dir]
+        use Adsf::Rack::IndexFileFinder,
+          root: site.config[:output_dir],
+          index_filenames: index_filenames
         run Rack::File.new(site.config[:output_dir])
       end.to_app

```

Greetings,
JMPC

**Comment by ddfreyne** [comment]
Makes sense! I prefer the former approach (the fewer configuration options, the better).

I’ve created #987 which adds a spec for the view command. I’ll follow up with a PR that adds support for `index.xhtml` in a bit.

**Comment by ddfreyne** [comment]
Implemented in #988. Thanks!

**Comment by ddfreyne** [comment]
FYI: This behavior will change to read the index filenames from the `index_filenames` site configuration attribute in 4.4.1. (See #989, #998)
